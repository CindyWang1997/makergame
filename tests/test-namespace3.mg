namespace A {
  object C { }
  object B {
    event create(C c) { std::print::s("create success"); }
    void compute(B b) { std::print::s("member success"); }
  }
  void compute(B b) { std::print::s("success"); }
}

object main {
  event create {
    A::B b = create A::B(create A::C);
    b.compute(b);
    A::compute(b);

    std::game::end();
  }
}
