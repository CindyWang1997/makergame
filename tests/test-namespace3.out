create success
member success
success
